attributes/default.rb

```diff
@@ -1 +1,18 @@
-default['sysfs']['package'] = "sysfsutils"
+default['sysfs']['package'] = 'sysfsutils'
+default['sysfs']['service'] = 'sysfsutils'
+
+default['sysfs']['init_type'] = value_for_platform(
```

Comment:
I meant something more along the lines of some helper functions detect the presence of systemd, upstart, and fall back to sysvinit:
```ruby
def systemd?
  IO.read('/proc/1/comm').chomp == 'systemd'
end

def upstart?
  File.executable?('/sbin/initctl'
end

def determine_init_subsytem
  if systemd?
    'systemd'
  elsif upstart?
    'upstart'
  else
    'initd'
  end
end
```
 We can still choose to overwrite it if we want (such as in a docker container where you would never run systemd) but we can try doing some sane default detection.